File: Solutions/78. Subsets/78. Subsets.go
func subsets(nums []int) [][]int {
    var res [][]int
    var tmp []int
    foo(nums,&res,tmp,0)
    return res
}
func foo(nums []int,res *[][]int,tmp []int,begin int){
    t:=make([]int,len(tmp))
    copy(t,tmp)
    (*res)=append(*res,t)
    for i:=begin;i<len(nums);i+=1{
        tmp=append(tmp,nums[i])
        foo(nums,res,tmp,i+1)
        tmp=tmp[:len(tmp)-1]
    }
}

File: Solutions/78. Subsets/78. Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        def foo(nums,res,tmp,begin):
            res.append(tmp.copy())
            for i in range(begin,len(nums)):
                tmp.append(nums[i])
                foo(nums,res,tmp,i+1)
                tmp.pop()
        res,tmp=[],[]
        foo(nums,res,tmp,0)
        return res